plugins/chanbackup.c

```diff
@@ -586,6 +586,84 @@ static struct command_result *json_your_peer_storage(struct command *cmd,
 
         return send_outreq(cmd->plugin, req);
 }
+
+static struct command_result *after_latestbkp(struct command *cmd,
+					        const char *buf,
+					        const jsmntok_t *params,
+					        void *cb_arg UNUSED)
+{
+        u64 version;
+	u32 timestamp;
+	struct scb_chan **scb;
+        struct json_stream *response;
+        const jsmntok_t *datastoretok = json_get_member(buf,
+                                                        params,
+                                                        "datastore");
+        struct out_req *req;
+
+        if (datastoretok->size == 0) {
+        	response = jsonrpc_stream_success(cmd);
+		json_add_string(response, NULL,
+                                "No backup received from peers");
+		return command_finished(cmd, response);
+        }
+
+        const jsmntok_t *bkp = json_get_member(buf,
+                                               json_get_arr(datastoretok, 0),
+                                               "hex");
+
+        u8 *res = tal_hexdata(cmd,
+                              json_strdup(cmd, buf, bkp),
+                              bkp->end - bkp->start);
+	if (!fromwire_static_chan_backup(cmd,
+                                         res,
+                                         &version,
+                                         &timestamp,
+                                         &scb)) {
+		plugin_err(cmd->plugin, "Corrupted SCB on disk!");
+	}
+
+	if (version != VERSION) {
+		plugin_err(cmd->plugin,
+                           "Incompatible version, Contact the admin!");
+	}
+
+        req = jsonrpc_request_start(cmd->plugin, cmd, "recoverchannel",
+				after_recover_rpc,
+				&forward_error, NULL);
+
+	json_array_start(req->js, "scb");
+	for (size_t i=0; i<tal_count(scb); i++) {
+		u8 *scb_hex = tal_arr(cmd, u8, 0);
+		towire_scb_chan(&scb_hex,scb[i]);
+		json_add_hex(req->js, NULL, scb_hex, tal_bytelen(scb_hex));
+	}
+	json_array_end(req->js);
+
+	return send_outreq(cmd->plugin, req);
+
+}
+
+static struct command_result *json_restorefrompeer(struct command *cmd,
+				      const char *buf,
+                                      const jsmntok_t *params)
+{
+	struct out_req *req;
+
+	if (!param(cmd, buf, params, NULL))
+		return command_param_failed();
+
+        req = jsonrpc_request_start(cmd->plugin,
+                                    cmd,
+                                    "listdatastore",
```

Comment:
OK, so I added more some datastore helpers in https://github.com/ElementsProject/lightning/pull/5809 (we already had ones to set datastore entries, which were included since you created this PR, and you should use those too).